apps/cloudbroker/base/cloudapi__disks/methodclass/cloudapi_disks.py

```diff
@@ -203,13 +203,12 @@ def resize(self, diskId, size, **kwargs):
             # Validate that enough resources are available in the CU limits to add the disk
             j.apps.cloudapi.accounts.checkAvailableMachineResources(disk.accountId, vdisksize=size)
 
-        provider = self.cb.getProviderByGID(disk.gid)
+        provider, _, _ = self.cb.getProviderAndNode(machine['id'])
```

Comment:
there is a chance machine is None
so I suggest you put this in the if machine and in the else keep the getProviderByGID
so the actually volume resize happens